ports/esp32/modnetwork.c

```diff
@@ -711,6 +734,11 @@ STATIC mp_obj_t esp_config(size_t n_args, const mp_obj_t *args, mp_map_t *kwargs
             val = MP_OBJ_NEW_SMALL_INT(cfg.ap.max_connection);
             break;
         }
+        case QS(MP_QSTR_reconnects):
+            req_if = WIFI_IF_STA;
+            int rec = conf_wifi_sta_reconnects - 1;
+            val = MP_OBJ_NEW_SMALL_INT(rec);
```

Comment:
I was thinking this might be able to return `wifi_sta_reconnects`, ie the number of reconnects left.  That could be useful information.  But one should also be able to query the value that is set, ie `conf_wifi_sta_reconnects`, and it makes more sense to return that value here.

(So that's just a comment, no change needed.)

Reply:
Even when we need this information (I can hardly imagine why), `wlan.config('reconnects')` doesn't look like the right place to return it. `.config` sounds like configuration for me, not a generic way to query arbitrary internal information from the driver.

Reply:
It could be `wlan.status('reconnects')`

Reply:
What would be better would be async events/callbacks when things change, eg wifi gets disconnected.  But that's definitely out of scope here.